lib/puppet/provider/cisco_vxlan_vtep_vni/nxapi.rb

```diff
@@ -0,0 +1,187 @@
+    current_state = {
+      name:   	  "#{interface} #{vni}",
+      interface: interface,
+      vni:       vni,
+      assoc_vrf: vni_instance.assoc_vrf,
+      ensure:    :present,
+      peer_ips:  vni_instance.peer_list,
+    }
+    # Call node_utils getter for each property
+    VXLAN_VTEP_VNI_NON_BOOL_PROPS.each do |prop|
+      current_state[prop] = vni_instance.send(prop)
+    end
+    VXLAN_VTEP_VNI_BOOL_PROPS.each do |prop|
+      val = vni_instance.send(prop)
+      if val.nil?
+        current_state[prop] = nil
+      else
+        current_state[prop] = val ? :true : :false
+      end
+    end
+    new(current_state)
+  end # self.properties_get
+
+  def self.instances
+    vnis = []
+    Cisco::VxlanVtepVni.vnis.each do |interface, all_vnis|
+      all_vnis.each do |vni, vni_instance|
+        vnis << properties_get(interface, vni, vni_instance)
+      end
+    end
+    vnis
+  end
+
+  def self.prefetch(resources)
+    vnis = instances
+    resources.keys.each do |id|
+      provider = vnis.find do |vni|
+        vni.interface.to_s == resources[id][:interface].to_s &&
+        vni.vni == resources[id][:vni] &&
+        vni.assoc_vrf.to_s == resources[id][:assoc_vrf].to_s
+      end
+      resources[id].provider = provider unless provider.nil?
+    end
+  end # self.prefetch
+
+  def exists?
+    (@property_hash[:ensure] == :present)
+  end
+
+  def create
+    @property_flush[:ensure] = :present
+  end
+
+  def destroy
+    @property_flush[:ensure] = :absent
+  end
+
+  def instance_name
+    vni
+  end
+
+  def properties_set(new_vni=false)
+    VXLAN_VTEP_VNI_ALL_PROPS.each do |prop|
+      next unless @resource[prop]
+      send("#{prop}=", @resource[prop]) if new_vni
+      unless @property_flush[prop].nil?
+        @vtep_vni.send("#{prop}=", @property_flush[prop]) if
+          @vtep_vni.respond_to?("#{prop}=")
+      end
+    end
+    # node utils is named peer_list= instead of peer_ips=
+    return unless @resource[:peer_ips]
+    self.peer_ips = @resource[:peer_ips] if new_vni
+    @vtep_vni.peer_list = @property_flush[:peer_ips] unless
+      @property_flush[:peer_ips].nil?
+  end
+
+  # can't autogen peer_ips, special array handling
+  def peer_ips
+    return [:default] if @resource[:peer_ips] &&
+                         @resource[:peer_ips][0] == :default &&
+                         @property_hash[:peer_ips] ==
+                         @vtep_vni.default_peer_list
+    @property_hash[:peer_ips]
+  end
+
+  def peer_ips=(set_value)
+    if set_value.is_a?(Array) && set_value[0] == :default
+      set_value = @vtep_vni.default_peer_list
+    end
+    @property_flush[:peer_ips] = set_value
+  end
+
+  def flush
+    if @property_flush[:ensure] == :absent
+      @vtep_vni.destroy
+      @vtep_vni = nil
+    else
+      # Create/Update
+      if @vtep_vni.nil?
+        new_vni = true
+        @vtep_vni = Cisco::VxlanVtepVni.new(@resource[:interface], @resource[:vni], @resource[:assoc_vrf])
```

Comment:
This won't work if `assoc_vrf` is a `TrueClass` object.  While I understand that puppet is passing this as a `Symbol` other orchestration frameworks that use this may not pass it as a symbol.  I would prefer that we leave this as `@assoc_vrf = assoc_vrf` and just make sure to pass the appropriate thing in puppet.  (https://github.com/cisco/cisco-network-puppet-module/pull/142/files#diff-1323926772cf8643b976e839a2d0f0fcR167)

```
assoc_vrf = @resource[:assoc_vrf] == :true ? true : false
@vtep_vni = Cisco::VxlanVtepVni.new(@resource[:interface], @resource[:vni], assoc_vrf)
```